source/Detector/Amptek/AmptekSDD123Server.cpp

```diff
@@ -0,0 +1,516 @@
+
+void AmptekSDD123Server::fakeSendDatagram(AmptekSDD123Packet packet, int overrideTimeout)
+{
+	if(socketLocallyBusy_){
+		qDebug() << "PACKET QUEUE IS BUSY (fake send datagram)";
+		packetQueue_.append(packet);
+	}
+	else{
+		socketLocallyBusy_ = true;
+		currentRequestPacket_ = packet;
+		currentRequestTime_.restart();
+		//udpDetectorSocket_->writeDatagram(packet.datagram(), packet.datagram().size(), configurationMap_->detectorAddress(), 10001);
+		connect(requestPacketTimer_, SIGNAL(timeout()), this, SLOT(requestDatagramTimeout()));
+		if(overrideTimeout == -1)
+			requestPacketTimer_->start(currentRequestPacket_.timeout());
+		else
+			requestPacketTimer_->start(overrideTimeout);
+	}
+}
+
+void AmptekSDD123Server::processPendingDatagrams()
+{
+	responsePacketReady_ = false;
+	while(udpDetectorSocket_->hasPendingDatagrams()){
+		QByteArray datagram;
+		bool ok;
+		datagram.resize(udpDetectorSocket_->pendingDatagramSize());
+		QHostAddress *sourceAddress = new QHostAddress();
+		udpDetectorSocket_->readDatagram(datagram.data(), datagram.size(), sourceAddress);
+
+		if(datagram.size() < 520 && datagram.mid(0,2) == QByteArray::fromHex("f5fa")){
+			totalResponseDatagramSize_ = datagram.size();
+			totalResponseDatagram_.clear();
+			totalResponseDatagram_.append(datagram);
+			responsePacketReady_ = true;
+			stillReceivingResponseDatagram_ = false;
+		}
+		else if(stillReceivingResponseDatagram_){
+			totalResponseDatagram_.append(datagram);
+			if(totalResponseDatagram_.size() == totalResponseDatagramSize_){
+				stillReceivingResponseDatagram_ = false;
+				responsePacketReady_ = true;
+			}
+		}
+		else if(datagram.size() == 520){
+			if(datagram.mid(0,2) == QByteArray::fromHex("f5fa")){
+				stillReceivingResponseDatagram_ = true;
+				totalResponseDatagramSize_ = datagram.mid(4,2).toHex().toInt(&ok, 16) + 8;//+8 for header, command, length, and checksum
+				totalResponseDatagram_.clear();
+				totalResponseDatagram_.append(datagram);
+			}
+		}
+		delete sourceAddress;
+	}
+	if(responsePacketReady_)
+		emit responsePacketReady();
+}
+
+void AmptekSDD123Server::onResponsePacketReady()
+{
+	requestPacketTimer_->stop();
+	syncPacketTimer_->stop();
+	AmptekSDD123Packet lastRequestPacket = currentRequestPacket_;
+	AmptekSDD123Packet responsePacket(currentRequestPacket_.packetID(), totalResponseDatagram_);
+
+	bool foundTimedOutPacket = false;
+	bool rerequestRequired = false;
+	if(timedOutPackets_.count() > 0){
+		for(int x = 0; x < timedOutPackets_.count(); x++){
+			qDebug() << timedOutPackets_.at(x).command() << QString(QByteArray::fromHex(timedOutPackets_.at(x).dataString().toAscii()));
+		}
+
+		AmptekSDD123Packet headPacket(-1, QString("NONE"));
+		while(!foundTimedOutPacket && timedOutPackets_.count() > 0){
+			headPacket = timedOutPackets_.dequeue();
+
+//			if(responsePacket.command() == "Comm test - Echo packet Response" && headPacket.command() == "Comm test - Echo packet" && responsePacket.dataString() == headPacket.dataString())
```

Comment:
There are a number of commented lines in the code below, also. Can these be removed?